File: backend/src/main/java/com/voyago/service/TransportService.java
package com.voyago.service;

import com.voyago.model.TravelOption;
import com.voyago.repository.TravelOptionRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class TransportService {

    private final TravelOptionRepository travelOptionRepository;

    public List<TravelOption> getAll() {
        return travelOptionRepository.findAll();
    }

    public List<TravelOption> getByMode(String mode) {
        return travelOptionRepository.findByMode(mode.toUpperCase());
    }

    public List<TravelOption> search(String fromCity, String toCity, String mode) {
        if (mode != null && !mode.isEmpty()) {
            return travelOptionRepository.findByFromCityAndToCityAndMode(fromCity, toCity, mode.toUpperCase());
        }
        return travelOptionRepository.findByFromCityAndToCity(fromCity, toCity);
    }

    public TravelOption getById(String id) {
        return travelOptionRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Travel option not found: " + id));
    }

    public TravelOption create(TravelOption option) {
        return travelOptionRepository.save(option);
    }

    public TravelOption update(String id, TravelOption updated) {
        TravelOption existing = getById(id);
        updated.setId(existing.getId());
        return travelOptionRepository.save(updated);
    }

    public void delete(String id) {
        travelOptionRepository.deleteById(id);
    }
}

File: src/services/transportService.ts
import apiClient from './api';
import { TravelOption } from '../types';

export const transportService = {
  async getAll(): Promise<TravelOption[]> {
    const { data } = await apiClient.get<TravelOption[]>('/transport');
    return data;
  },

  async search(from: string, to: string, mode?: string): Promise<TravelOption[]> {
    const { data } = await apiClient.get<TravelOption[]>('/transport/search', {
      params: { from, to, ...(mode ? { mode } : {}) },
    });
    return data;
  },

  async getByMode(mode: string): Promise<TravelOption[]> {
    const { data } = await apiClient.get<TravelOption[]>(`/transport/mode/${mode}`);
    return data;
  },

  async getById(id: string): Promise<TravelOption> {
    const { data } = await apiClient.get<TravelOption>(`/transport/${id}`);
    return data;
  },

  async create(option: Omit<TravelOption, 'id'>): Promise<TravelOption> {
    const { data } = await apiClient.post<TravelOption>('/transport', option);
    return data;
  },

  async update(id: string, option: Partial<TravelOption>): Promise<TravelOption> {
    const { data } = await apiClient.put<TravelOption>(`/transport/${id}`, option);
    return data;
  },

  async delete(id: string): Promise<void> {
    await apiClient.delete(`/transport/${id}`);
  },
};

export default transportService;
